test/e2e/typescript-version-warning/typescript-version-warning.test.ts

```diff
@@ -0,0 +1,27 @@
+import { createNextDescribe } from 'e2e-utils'
+
+createNextDescribe(
+  'typescript-version-warning',
+  {
+    files: __dirname,
+    skipStart: true,
+    skipDeployment: true,
+    dependencies: {
+      typescript: '4.0.6',
+    },
+  },
+  ({ next, isNextDeploy, isNextDev }) => {
+    if (isNextDeploy || isNextDev) {
+      it('should skip', async () => {
+        expect(1).toBe(1)
+      })
+    } else {
+      it('should print warning when old typescript version is used with next build', async () => {
+        await next.start().catch(() => {})
+        expect(next.cliOutput).toContain(
+          'warn Minimum recommended TypeScript version is v4.5.2, older versions can potentially be incompatible with Next.js. Detected: 4.0.6'
+        )
+      })
+    }
```

Comment:
```suggestion
    if (isNextDeploy || isNextDev) {
      it('should skip', () => {})
      return
    }

    it('should print warning when old typescript version is used with next build', async () => {
      await next.start().catch(() => {})
      expect(next.cliOutput).toContain(
        'warn Minimum recommended TypeScript version is v4.5.2, older versions can potentially be incompatible with Next.js. Detected: 4.0.6'
      )
    })
```

Reply:
Fixed in [81c0f18](https://github.com/vercel/next.js/pull/50557/commits/81c0f18964d450b1df7576d69d9a2931307e0de7)